nomos-services/utils/src/overwatch/status/service_status_entry/error.rs

```diff
@@ -0,0 +1,40 @@
+//! Provides conversion from a collection of [`ServiceStatusEntry`] to
+//! the general-purpose [`DynError`](overwatch::DynError) type.
+
+use std::fmt::{Debug, Display, Formatter};
+
+use crate::overwatch::status::ServiceStatusEntry;
+
+#[derive(Debug)]
+pub struct ServiceStatusEntriesError<RuntimeServiceId: Display> {
```

Comment:
I think we don't need the `Display` bound at this point but only in the impl block.

Reply:
It's required by `std::error:Error`.

Reply:
Yes but we can require `Display` only when needed, no? Why does it have to be in the struct definition?